torch/csrc/jit/codegen/cuda/fusion.cpp

```diff
@@ -614,9 +614,6 @@ std::vector<Val*> Fusion::getTerminatingOutputs() {
 }
 
 void Fusion::aliasOutputToInput(Val* output, Val* input) {
-  TORCH_INTERNAL_ASSERT(
-      hasInput(input) && hasOutput(output),
-      "alias only allows from output to input");
```

Comment:
Would we need to check if `input` is an alias too? I know it wouldn't happen in the intended use case but just wanted to be sure.

Reply:
> Just curious, why do we want to remove this assert?

Christian hated the idea of having premature checks that can't be enforced later. e.g. input/output could be erased from fusion, so this check doesn't mean much here.

> input is an alias too? 

aliased input shouldn't be too much of a concern for us in constructing fusion parameters, since inputs are given to executor, but not allocated by executors.
So the only implication for fusion with an aliased input, is to check for racing/dependency violation, which we are not doing in codegen yet.
I think the same thing would apply to aliasing to intermediate tensors. One thing that we might need are outputs aliasing each other, I brought it up when we discuss `autocast_to_fpXX` but it got shot down :)
I'll open an issue to track that topic and I'll make sure to keep it general enough that we can carry this conversation over as well.